src/algorithms/cDIIS.jl

```diff
@@ -5,17 +5,30 @@
 """
     Container type for the state of one spin type for cDIIS
 """
-mutable struct cDIISstate
-    fock::CircularBuffer
+mutable struct cDIISstate  # TODO Rename to DiisState
+    fock::CircularBuffer   # TODO Rename to iterate
     error::CircularBuffer
+
+    # TODO Describe the structure of errorOverlaps here already
     errorOverlaps::CircularBuffer
+    # TODO Put n_diis_size in here
```

Comment:
I'm not too sure I understand you correctly.

Do you want DiisState to be elements of the Circular Buffer or let a single state contain several Circular Buffers?
In the former case: Do I assume correctly, that you would like to have the separation by spin in the cDIIS object?